WinRTXamlToolkit.Controls.DataVisualization.UWP/OrientedPanel.cs

```diff
@@ -1,680 +1,697 @@
-        /// length selector.
-        /// </summary>
-        /// <param name="children">A sequence of children.</param>
-        /// <param name="lengthSelector">A function that returns a length given
-        /// a UIElement.</param>
-        /// <returns>A sequence of ranges.</returns>
-        private static IEnumerable<Range<double>> GetRanges(IEnumerable<UIElement> children, Func<UIElement, double> lengthSelector)
-        {
-            return 
-                children
-                    .Select(child =>
-                    {
-                        double centerCoordinate = GetCenterCoordinate(child);
-                        double halfLength = lengthSelector(child) / 2;
-                        return new Range<double>(centerCoordinate - halfLength, centerCoordinate + halfLength);
-                    });
-        }
-
-        /// <summary>
-        /// Measures children and determines necessary size.
-        /// </summary>
-        /// <param name="availableSize">The available size.</param>
-        /// <returns>The necessary size.</returns>
-        [SuppressMessage("Microsoft.Maintainability", "CA1502:AvoidExcessiveComplexity", Justification = "Linq use artificially increases cyclomatic complexity.  Linq functions are well-understood.")]
-        protected override Size MeasureOverride(Size availableSize)
-        {
-            double offset = 0.0;
-
-            if (Children.Count > 0)
-            {
-                Size totalSize = new Size(double.PositiveInfinity, double.PositiveInfinity);
-                foreach (UIElement child in this.Children)
-                {
-                    child.Measure(totalSize);
-                }
-
-                Func<UIElement, double> lengthSelector = null;
-                Func<UIElement, double> offsetSelector = null;
-
-                if (Orientation == Orientation.Horizontal)
-                {
-                    lengthSelector = child => GetCorrectedDesiredSize(child).Width;
-                    offsetSelector = child => GetCorrectedDesiredSize(child).Height;
-                }
-                else
-                {
-                    lengthSelector = child => GetCorrectedDesiredSize(child).Height;
-                    offsetSelector = child => GetCorrectedDesiredSize(child).Width;
-                }
-
-                IEnumerable<IGrouping<int, UIElement>> priorityGroups =
-                    from child in Children
-                    group child by GetPriority(child) into priorityGroup
-                    select priorityGroup;
-
-                ActualMinimumDistanceBetweenChildren =
-                    (from priorityGroup in priorityGroups
-                     let orderedElements =
-                         (from element in priorityGroup
-                          orderby GetCenterCoordinate(element) ascending
-                          select element).ToList()
-                     where orderedElements.Count >= 2
```

Comment:
this `where` right here seems to be the source of the bug.  if all of the groups have just one item (which seems to be a timing issue, at least for me) , then all of the groups get filtered out here, and the min throws because there are no elements.

also considered adding an extension method like `MinOrDefault(0)` that would return a default value if a sequence was empty.